src/memory/smarteeprom.c

```diff
@@ -55,6 +56,14 @@ void smarteeprom_disable(void)
 
 void smarteeprom_setup(void)
 {
+    /*
+     * First, erase the last blocks of flash.
+     * So we are sure that there is no garbage data
+     * in them.
+     */
+    if (!memory_cleanup_smarteeprom()) {
+        Abort("Failed to erase SmartEEPROM memory area.");
```

Comment:
I prefer if we don't abort in the boot path.

Reply:
Fixed in e11e471